src/main/java/Duke.java

```diff
@@ -1,10 +1,181 @@
+                ProcessList(counter, list);
+            } else if (input.contains("done") ) { //check if done
+                ProcessDone(input, list);
+            } else {
+                counter = ProcessTasks(input, counter, list); //process tasks
+            }
+        }
+    }
+
+    private static void CheckIsEmptyList(Task[] list) {
+        boolean isEmptyList;
+        if(list[0] != null) {
+            isEmptyList = false;
+        }
+    }
+
+    private static void ProcessList(int counter, Task[] list) {
+        if (list[0] != null){
+            PrintListMessage(counter, list);
+        } else {
+            PrintListButEmptyMessage();
+        }
+    }
+
+    private static void ProcessDone(String input, Task[] list) {
+        int donePos = input.indexOf("done");
+        if (list[0] != null) {
+            if (input.length() < donePos + 5) {
+                PrintDoneButNotSpecificMessage(); //when input contains done but no number
+            } else {
+                ProcessDone(input, list, donePos); //when input contains done and specified number
+            }
+        } else {
+            PrintDoneButEmptyMessage(); //when input contains done but list is empty
+        }
+    }
+
+    private static void ProcessDone(String input, Task[] list, int donePos) {
+        String itemNumDone = input.substring(donePos + 5, donePos + 6);
+        int itemNum = Integer.parseInt(itemNumDone);
+        list[itemNum - 1].setDone();
+        PrintDoneMessage(list, itemNum);
+    }
+
+    private static int ProcessTasks(String input, int counter, Task[] list) {
+        if (input.contains("todo")) {
+            String description = input.substring(5);
+            ToDo newTask = new ToDo(description);
+            list[counter] = newTask;
+            counter += 1;
+            PrintAddedTaskMessage(newTask, counter);
+        } else if (input.contains("deadline")) {
+            int donePos = input.indexOf("/"); //finds pos of '/'
+            String description = input.substring(9,donePos);
+            String date = input.substring(donePos + 4);
+            Deadline newTask = new Deadline(description,date);
+            list[counter] = newTask;
+            counter += 1;
+            PrintAddedTaskMessage(newTask, counter);
+        } else if (input.contains("event")) {
+            int donePos = input.indexOf("/"); //finds pos of '/'
+            String description = input.substring(6,donePos);
+            String date = input.substring(donePos + 4);
+            Event newTask = new Event(description,date);
+            list[counter] = newTask;
+            counter += 1;
+            PrintAddedTaskMessage(newTask, counter);
+        } else {
+            System.out.println("Please specify tasks: todo, deadline or event");
+            System.out.println("Example - type in the following: todo read book");
+        }
+        return counter;
+    }
+
+    private static void PrintDoneMessage(Task[] list, int itemNum) {
```

Comment:
Nice use of specific PrintMessage methods to keep the overall code cleaner and easier to understand